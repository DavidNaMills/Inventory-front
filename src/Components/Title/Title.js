import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);