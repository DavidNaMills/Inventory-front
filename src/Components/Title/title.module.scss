@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}