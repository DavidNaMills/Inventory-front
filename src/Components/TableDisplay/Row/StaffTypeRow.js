import React from 'react';
import { isTablet, withOrientationChange } from 'react-device-detect';
import Title from '../../Title/Title';
import classes from './ProductTypeRow/ProductTypeRow.module.scss';
import { useLocale } from '../../../Context/LocaleContext';

const StaffTypeRow = (props) => {
    const locale = useLocale();
    const { config, data, withTitle = true, showInfo=true } = props;

    const isRole = (key, txt) =>{
        if(key === 'role'){
            return locale[txt];
        } else {
            return txt;
        }
    }

    return (
        <div className={classes.productTypeRow}>
            {withTitle &&
                <div className={classes.productTypeRow__title}>
                    <Title message={data[config.main[0]]} size='h3' />
                </div>
            }
            {showInfo &&
                <div className={(isTablet && props.isPortrait) ? classes.productTypeRow__cellContPortrait : classes.productTypeRow__cellCont}>
                    <p><span>{locale[config.main[1]]}:</span>{isRole(config.main[1], data[config.main[1]])}</p>
                    <p><span>{locale[config.main[2]]}:</span>{isRole(config.main[2], data[config.main[2]])}</p>
                </div>
            }
        </div>
    )
}

export default withOrientationChange(StaffTypeRow);