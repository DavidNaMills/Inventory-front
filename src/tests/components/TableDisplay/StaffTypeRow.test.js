import React from 'react';
import '../../testHelpers/context';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import * as device from 'react-device-detect';

import StaffTypeRow from '../../../Components/TableDisplay/Row/StaffTypeRow';
import Title from '../../../Components/Title/Title';

const baseProps = {
    config: {main : ['name', 'code', 'qtyInStock']},
    data: {
        name: 'Blue dogs',
        location: 'Dundee',
        qtyInStock: '12345678',
        code: '187wertt56',
        url: 'test.jpg'
    }
}


describe('<StaffTypeRow', ()=>{
    it('matches snapshot', ()=>{
        const comp = renderer.create(<StaffTypeRow {...baseProps} />)
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('exists', ()=>{
        const wrapper = shallow(<StaffTypeRow {...baseProps}/>);
        expect(wrapper.exists).toBeTruthy();
    });


    describe('isMobileOnly === false tests', ()=>{
        let wrapper;
        const orig = device.isMobileOnly;

        beforeEach(()=>{
            device.isMobileOnly = false;
            wrapper = mount(<StaffTypeRow {...baseProps}/>);
        });

        afterAll(()=>{
            device.isMobile = orig;
        });

        it('renders 3 <div> elements with classNames: productTypeRow, productTypeRow__title, productTypeRow__cellCont', ()=>{
            const div = wrapper.find('div');

            expect(div.length).toBe(4);
            expect(div.at(0).props()).toHaveProperty('className', 'productTypeRow');
            expect(div.at(1).props()).toHaveProperty('className', 'productTypeRow__title');
            expect(div.at(3).props()).toHaveProperty('className', 'productTypeRow__cellCont');
        });
        
        it('renders a <Title/> component', ()=>{
            const ttl = wrapper.find(Title);
            expect(ttl.length).toBe(1);
            expect(ttl.props()).toHaveProperty('message', baseProps.data[baseProps.config.main[0]]);
            expect(ttl.props()).toHaveProperty('size', 'h3');
        });
        
        it('renders 2 <p> elements with text', ()=>{
            const p = wrapper.find('p');
            expect(p.length).toBe(2);
            expect(p.at(0).text()).toEqual('code:187wertt56');
            expect(p.at(1).text()).toEqual('qty available:12345678');
        });

        // FIXME: how to override || inject properties from a HOC
        describe.skip('isTablet && isPortait both equal true tests', ()=>{
            let orig2;
            
            beforeEach(()=>{ 
                orig2 = device.isTablet; 
            });
            afterEach(()=>{ 
                device.isTablet = orig2;
             });

            it('sets the 3rd <div> className to productTypeRow__cellContPortrait if isTablet && isPortait equal true', ()=>{
                device.isTablet = true;
                
                const tW = mount(<StaffTypeRow {...baseProps} isPortrait={true}/>);
                const div = tW.find('div');
                expect(div.length).toBe(4);
                expect(div.at(3).props()).toHaveProperty('className', 'productTypeRow__cellContPortrait');
            });
            
            it('does not change the 3rd <div> classname if isTablet is true but in landscape. classname: productTypeRow__cellCont', ()=>{
                device.isTablet = true;
                const tW = mount(<StaffTypeRow {...baseProps} isPortrait={false}/>);
                const div = tW.find('div');
                expect(div.length).toBe(4);
                expect(div.at(3).props()).toHaveProperty('className', 'productTypeRow__cellCont');
            });
            
            it('does not change the 3rd <div> classname if isTablet is false but in portrait. classname: productTypeRow__cellCont', ()=>{
                device.isTablet = false;
                const tW = mount(<StaffTypeRow {...baseProps} isPortrait={true}/>);
                const div = tW.find('div');
                expect(div.length).toBe(4);
                expect(div.at(3).props()).toHaveProperty('className', 'productTypeRow__cellCont');
            });
        });
    });

    describe('isMobileOnly === true tests', ()=>{
        let wrapper;
        const orig = device.isMobileOnly;

        beforeEach(()=>{
            device.isMobileOnly = true;
            wrapper = mount(<StaffTypeRow {...baseProps} showInfo={false}/>);
        });

        afterAll(()=>{
            device.isMobile = orig;
        });

        it('only renders 4 <div> elements with classnames: productTypeRow, productTypeRow__title. 1 div part of <Title/>', ()=>{
            const div = wrapper.find('div');
            expect(div.length).toBe(3);
            expect(div.at(0).props()).toHaveProperty('className', 'productTypeRow');
            expect(div.at(1).props()).toHaveProperty('className', 'productTypeRow__title');
        });
        
        it('renders a <Title/> component', ()=>{
            const ttl = wrapper.find(Title);
            expect(ttl.length).toBe(1);
            expect(ttl.props()).toHaveProperty('message', baseProps.data[baseProps.config.main[0]]);
            expect(ttl.props()).toHaveProperty('size', 'h3');
        });

        it('does not render any <p> elements or information', ()=>{
            expect(wrapper.find('.productTypeRow__cellContPortrait').length).toBe(0);
            expect(wrapper.find('.productTypeRow__cellCont').length).toBe(0);
            expect(wrapper.find('p').length).toBe(0);
        });
    });
});