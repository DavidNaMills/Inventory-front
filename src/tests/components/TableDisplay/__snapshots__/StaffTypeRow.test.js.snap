// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffTypeRow matches snapshot 1`] = `
<div
  className="productTypeRow"
>
  <div
    className="productTypeRow__title"
  >
    <div>
      <h3
        className="title_h3 "
      >
        Blue dogs
      </h3>
    </div>
  </div>
  <div
    className="productTypeRow__cellCont"
  >
    <p>
      <span>
        Code
        :
      </span>
      187wertt56
    </p>
    <p>
      <span>
        Qty
        :
      </span>
      12345678
    </p>
  </div>
</div>
`;
